packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/WindowUtil.kt

```diff
@@ -56,8 +56,8 @@ private fun Window.statusBarHide() {
 }
 
 @Suppress("DEPRECATION")
-private fun Window.statusBarShow() {
-  if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+private fun Window.statusBarShow(isEdgeToEdge: Boolean) {
+  if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R && !isEdgeToEdge) {
```

Comment:
Always was hesitant to ask, but now I feel like it's a right time 🙂 

From what I remember `StatusBar` management uses some deprecated methods for managing its properties. As a result it may break some animations driven by ` WindowInsetsAnimationCompat.Callback`.

To overcome this problem I had to create own [module](https://github.com/kirillzyusko/react-native-keyboard-controller/blob/main/android/src/main/java/com/reactnativekeyboardcontroller/modules/StatusBarManagerCompatModuleImpl.kt) based on `WindowInsetsControllerCompat` class and monkey-patch JS module afterward.

This is directly not related to this PR, but I'd like to discuss a possibility to rewrite StatusBar management to a new/modern API 🙌 (whether FB team is open to it and if not then what blocks/prevents that).